static/js/image_upload_widget.js

```diff
@@ -44,18 +46,30 @@ exports.set_up = function () {
             success() {
                 image_upload_complete(spinner, upload_text, delete_button);
                 error_field.hide();
+                if (widget === "user-avatar-upload-widget") {
+                    $("#user-avatar-source").hide();
+                }
             },
             error(xhr) {
                 image_upload_complete(spinner, upload_text, delete_button);
+                if (widget === "user-avatar-upload-widget") {
+                    if (page_params.avatar_source === "G") {
+                        $("#user-avatar-source").show();
+                    }
+                }
                 ui_report.error("", xhr, error_field);
             },
         });
     }
-
-    realm_icon.build_realm_icon_widget(upload_image, null, true);
-    if (page_params.zulip_plan_is_not_limited) {
-        realm_logo.build_realm_logo_widget(upload_image, false);
-        realm_logo.build_realm_logo_widget(upload_image, true);
+    if (section === "your-account") {
+        avatar.build_user_avatar_widget(upload_image);
+    }
+    if (section === "org_misc") {
+        realm_icon.build_realm_icon_widget(upload_image, null, true);
+        if (page_params.zulip_plan_is_not_limited) {
+            realm_logo.build_realm_logo_widget(upload_image, false);
```

Comment:
Going ahead with the constructor idea, we can pass multiple options to the widget like:

```
const realm_icon_upload_widget = image_upload_widget({
	name: 'realm-icon',
	enabled: page_params.zulip_plan_is_not_limited,
});
```